### PR TITLE
[Staking]: add ETH pooled staking tests

### PR DESCRIPTION
**Pull request by miloserdow**
## Description

Add tests for ETH staking tx using RocketPool (see #2674).

## How to test

Run unit tests

## Types of changes

* New feature (non-breaking change which adds functionality)

## Checklist

- [X] Create pull request as draft initially, unless its complete.
- [X] Add C++ tests.
- [x] Add Kotlin tests.
- [x] Add Swift tests.

### CODE REVIEW

File: tests/chains/Ethereum/TWAnySignerTests.cpp [hunk 1]
```diff
@@ -485,4 +487,93 @@ TEST(TWAnySignerEthereum, SignERC1155Transfer_1559) {
...
+TEST(TWAnySignerEthereum, StakeRocketPool) {
+    auto chainId = store(uint256_t(1));
+    auto nonce = store(uint256_t(1));
+    auto gasPrice = store(uint256_t(2002000000));
+    auto gasLimit = store(uint256_t(205000));
+    auto maxFeePerGas = store(uint256_t(27900000000));
+    auto maxInclusionFeePerGas = store(uint256_t(1000000000));
+
+    auto toAddress = "0x2cac916b2a963bf162f076c0a8a4a8200bcfbfb4";
+    auto valueData = parse_hex("2386f26fc10000");
```
Comment (hewigovens) [inline tests/chains/Ethereum/TWAnySignerTests.cpp:499]: I think you can use ABI api to create this data, much more readable

File: tests/chains/Ethereum/TWAnySignerTests.cpp [hunk 2]
```diff
@@ -485,4 +487,93 @@ TEST(TWAnySignerEthereum, SignERC1155Transfer_1559) {
...
+    auto chainId = store(uint256_t(1));
+    auto nonce = store(uint256_t(3));
+    auto gasPrice = store(uint256_t(2002000000));
+    auto gasLimit = store(uint256_t(350000));
+    auto maxFeePerGas = store(uint256_t(27900000000));
+    auto maxInclusionFeePerGas = store(uint256_t(1000000000));
+
+    auto toAddress = "0xae78736Cd615f374D3085123A210448E74Fc6393";
+    auto valueData = parse_hex("0");
+    auto data = parse_hex("42966c680000000000000000000000000000000000000000000000000021faa32ab2502b");
```
Comment (hewigovens) [inline tests/chains/Ethereum/TWAnySignerTests.cpp:544]: same ABI remark here

**Review by Milerius — COMMENTED** [review]
Would be nice to add also a unit test for unstaking

**Comment by miloserdow** [comment]
> Would be nice to add also a unit test for unstaking

Already added

**Comment by Milerius** [comment]
> > Would be nice to add also a unit test for unstaking
> 
> Already added

yes sorry, is the rewards distributed automatically or user have to claim it automatically?

**Comment by miloserdow** [comment]
> yes sorry, is the rewards distributed automatically or user have to claim it automatically?

They are distributed automatically (rETH/ETH exchange price increases).

**Comment by miloserdow** [comment]
Android test failed because of formating issue, the results of both tests are correct

**Review by hewigovens — APPROVED** [review]

**Pull request merged**